### PR TITLE
Skip ring of experts all gather when EP = 1

### PR DESCRIPTION
**Pull request by Shuwen-Fang**
# Description

Currently runs error out when EP = 1 and ring of experts is used, fix by skipping all gather when EP = 1. This PR updates the logic to only all gather when EP > 1.

# Tests

CI tests

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/maxtext/configs/base.yml [hunk 1]
```diff
@@ -622,7 +622,7 @@ ici_autoregressive_parallelism: 1
 ici_pipeline_parallelism: 1
 ici_expert_parallelism: 1
 
-# Enabling check_vma is recommended for improved performance. Only supported for EP / FSDP ICI parallelisms, shard_mode: "auto", use_ragged_sort: False, use_ring_of_experts: False, and use_tokamax_gmm=False.
+# Enabling check_vma is recommended for improved performance. Only supported for EP / FSDP ICI parallelisms, shard_mode: "auto", use_ragged_sort: False, and use_tokamax_gmm=False.
```
Comment (NuojCheng) [inline src/maxtext/configs/base.yml:625]: emmm why this flag is removed?

File: src/maxtext/layers/moe.py [hunk 2]
```diff
@@ -1364,12 +1364,12 @@ def route(x, logits, pre_bias_logits, rngs):
...
-        x, logits, pre_bias_logits = tuple(
-            jax.lax.all_gather(z, axis_name=self._expert_parallelism_name, tiled=True)
-            for z in (x, logits, pre_bias_logits)
-        )
+        if num_ep != 1:
+          # Duplicate inputs to all expert shards.
+          x, logits, pre_bias_logits = tuple(
+              jax.lax.all_gather(z, axis_name=self._expert_parallelism_name, tiled=True)
+              for z in (x, logits, pre_bias_logits)
+          )
```
Comment (NuojCheng) [inline src/maxtext/layers/moe.py:1372]: I think we should error out if `use_ring_of_experts=true` and `num_ep==1`
